lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -3325,7 +3325,7 @@ constructSharedTransaction
                         Just (ApiPaymentAddresses content) ->
                             F.toList (addressAmountToTxOut <$> content)
                 (unbalancedTx, scriptLookup) <- liftHandler $
-                    W.constructUnbalancedSharedTransaction @n era
+                    W.constructUnbalancedSharedTransaction @n @era
```

Comment:
```suggestion
                    W.constructUnbalancedSharedTransaction @n
```

Reply:
@HeinrichApfelmus If I've interpreted your comment correctly, you're proposing to replace this:

```hs
(unbalancedTx, scriptLookup) <- liftHandler $
    W.constructUnbalancedSharedTransaction @n @era
    db txCtx PreSelection {outputs = outs}
```

with (something like):

```hs
(unbalancedTx :: Cardano.TxBody (Write.CardanoApiEra era)
    , scriptLookup
    ) <- liftHandler $
        W.constructUnbalancedSharedTransaction @n
        db txCtx PreSelection {outputs = outs}
```

Of course, there are probably nicer ways to format this (so that we don't need a multi-line tuple).

But I'm not sure what we gain by replacing the type application (on the function) with a type annotation (on the result) here.

Reply:
Same argument as for the previous comment.

The type annotation for `unbalancedTx` adds documentation — in this case, telling us in which `era` this unbalanced transaction is contructed (namely the one in scope).

In contrast, the type application does not add documentation, as one still has to inspect the type of `constructUnbalancedSharedTransaction @n @era` in order to compute the type of `unbalancedTx`.

Reply:
> In contrast, the type application does not add documentation, as one still has to inspect the type of constructUnbalancedSharedTransaction @n @era in order to compute the type of unbalancedTx.

Thanks.

Though I disagree with the notion that the `@era` type application adds no documentation.

When I read this code, I see the `@era` type application as documentation of two facts:
1. That the construction of a transaction is era-dependent.
2. That we're constructing an (unbalanced) transaction for the `era` type that is currently in scope.

Reply:
> When I read this code, I see the @era type application as documentation of two facts:
> 1. That the construction of a transaction is era-dependent.
> 2. That we're constructing an (unbalanced) transaction for the era type that is currently in scope.

Both of these items depend on knowledge of the type of `constructUnbalancedSharedTransaction`. In principle, the implementation of this function could choose to ignore the type argument.

In contrast, the type signature on `unbalancedTx` provides compiler-checked documentation that the result of `constructUnbalancedSharedTransaction` indeed satisfies these two facts. The facts being documentation are the same, but the type signature is superior in that

* it pertains to the *result* of `constructUnbalancedSharedTransaction`, not to the *arguments*

The point of the type signature on `unbalancedTx` is that these two facts are both documented and compiler-checked, without the reader needing to consult the implementation or even the type of `constructUnbalancedSharedTransaction`.